sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -124,6 +124,52 @@ struct snd_soc_ops max_98373_ops = {
 };
 EXPORT_SYMBOL_NS(max_98373_ops, SND_SOC_INTEL_SOF_MAXIM_COMMON);
 
+/*
+ * Find present Maxim codecs on board
+ *
+ * amp_nums: return the codec numbers, present 2 or 4 codecs support.
+ */
+
+int get_maxim_num_codec(void)
```

Comment:
How about passing device name ("MX98373" or "MX98390") into get_maxim_num_codec()? So that you don't need to check both "MX98373" and "MX98390"